relay-server/src/services/project/state.rs

```diff
@@ -114,3 +114,13 @@ pub struct LimitedParsedProjectState {
     #[serde(flatten)]
     pub info: ProjectInfo,
 }
+
+/// Response indicating whether a project state needs to be updated
+/// or the upstream does not have a newer version.
+#[derive(Debug, Clone)]
+pub enum UpstreamProjectState {
+    /// The upstream sent a [`ProjectState`].
+    New(ProjectState),
+    /// The upstream indicated that there is no newer version of the state available.
+    Unchanged,
```

Comment:
mini nit: We could call this `NotModified` to match [HTTP jargon](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304).